test/STM32CubeIDE/Application/User/ThrottlingController.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * ThrottlingRecord.cpp
+ *
+ *  Created on: 20. srp 2022.
+ *      Author: Luka
+ */
+
+#include "ThrottlingController.hpp"
+
+
+extern "C" uint8_t getThrottling (uint8_t eventType);
+extern "C" void timeForThrottling (uint32_t time);
+/*
+ThrottlingController::updateTimes(uint32_t inc) {
+}
+*/
+
+ThrottlingController tc;
+
+ThrottlingController::ThrottlingController(){
+
+}
+
+bool ThrottlingController::shuldThrottle(Event::Type eventType) {
+	bool throttling = throttlingRecords[eventType].shuldThrottle();
```

Comment:
Ovo ti je sad ordered lista tj. event je index, to je OK implementacija al imaš getEventType metodu pomoću koje možeš ovdje loopati dok ne dođeš do odgovarajućeg eventa. Tako možeš onda imati generički EventType, I ne moraš paziti na redosljed kojim dodaješ evente u array...

Reply:
riješeno